### ISSUE TITLE
Example `drop.rs` segfaults

### ISSUE THREAD

**Issue by ShivanshuKantPrasad** [raylib-rs/raylib-rs#40]
Running example `drop.rs` results in a segfault. The segfault seems to occur in `UnloadShader` but I can't figure out why. It happens in both `master` and `5.0.0` branch.

**Comment by ShivanshuKantPrasad** [comment]
This minimum example seg faults 
``` rust
use raylib::prelude::*;

fn main() {
    let _shader = {
        let (mut rl, thread) = raylib::init().size(400, 400).title("test").build();
        rl.load_shader(&thread, None, Some("static/shader/pbr.fs"))
            .expect("shader didn't load")
    };
}
```

but this minimum example runs fine
``` rust
use raylib::ffi::*;

fn main() {
    unsafe {
        InitWindow(400, 400, "test".as_ptr() as *const i8);
        let sh = LoadShader(std::ptr::null(), "static/shader/pbr.fs".as_ptr() as *const i8);
        CloseWindow();
        UnloadShader(sh);
    }
}
```
I assume the problem is with the higher level api somehow.

**Comment by ShivanshuKantPrasad** [comment]
This minimum C code works
``` c
#include <raylib.h>

int main() {
  InitWindow(400, 400, "Hello");
  Shader sh = LoadShader(0, "pbr.fs");
  CloseWindow();
  UnloadShader(sh);
  return 0;
}
```

but this minimum rust code doesn't work
```rust
use raylib::ffi::*;

fn main() {
    unsafe {
        InitWindow(400, 400, "test".as_ptr() as *const i8);
        let sh = LoadShader(
            std::ptr::null(),
            std::ffi::CString::new("static/shader/pbr.fs")
                .unwrap()
                .as_ptr(),
        );
        CloseWindow();
        UnloadShader(sh);
    }
}
```

My understanding is that the above code are completely identical, since it is directly calling the underlying c code with the ffi bindings.

Looking through the raylib source code, it seems `CloseWindow` calls `ClosePlatform` which calls `glfwTerminate` and `glfwDestroyWindow` which destroys the OpenGL context. Calling UnloadShader after the OpenGL context is destroyed shouldn't work. I believe the error is in the c code.

Is shader being droppable after the window is closed actually the desired behaviour?

**Comment by ShivanshuKantPrasad** [comment]
Running the previous C example in a vm caused it to seg fault. I am gonna assume that it is the expected behaviour and that Shaders aren't supposed to live longer than the window. I am gonna remove the `test_shader_drop` function from the `drop.rs` example.

**Comment by ShivanshuKantPrasad** [comment]
Fixed by #39
